From 8d8715ad055f426869d7ab9f21ec4e47844e2530 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 19 Jan 2022 07:54:32 +0200
Subject: [PATCH] [SW-58071] arch: common: move spi limit function device specific

This patch moves spi limit write function to device specific as
spi limit configuration is required for each device with different
values.

Change-Id: Ib990e0911a95482e0ae1f6b5294d96805db9799f
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index 025b122..a173502 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -116,11 +116,6 @@
 	security_disable_jtag_if();
 }
 
-__attribute__((weak)) void arcboot_rom_set_spi_write_limit(void)
-{
-	regmap_writel(write_en_16MB, mmPSOC_SECURITY_SPI_WRITE_LIMIT);
-}
-
 static void arcboot_rom_init(void)
 {
 	uint32_t reg;
diff --git a/arch/greco/arc_elf32/common.c b/arch/greco/arc_elf32/common.c
index 9c034d9..6a7b738 100644
--- a/arch/greco/arc_elf32/common.c
+++ b/arch/greco/arc_elf32/common.c
@@ -11,6 +11,7 @@
 #include <psoc_efuse_regs.h>
 #include <psoc_global_conf_masks.h>
 #include <arcboot.h>
+#include <spi_flash.h>
 
 #define IATU_BASE				(mmPCIE_DBI_BASE & 0xFFFFFF)
 #define IATU_REGION_CTRL_1_OFF_INBOUND		(IATU_BASE + 0x100)
@@ -167,3 +168,9 @@
 {
 	/* Do this after boot rom code, since it is beyond TO */
 }
+
+void arcboot_rom_set_spi_write_limit(void)
+{
+	regmap_writel(write_en_16MB, mmPSOC_SECURITY_SPI_WRITE_LIMIT);
+}
+
diff --git a/common/common.h b/common/common.h
index 3e2d944..9c7fd01 100644
--- a/common/common.h
+++ b/common/common.h
@@ -13,5 +13,6 @@
 void arcboot_rom_handle_tpm(void);
 void arcboot_rom_request_cpld_wdog(void);
 void arcboot_rom_clear_sw_status(void);
+void arcboot_rom_set_spi_write_limit(void);
 
 #endif /* _COMMON_H_ */
